tools/xnvme.c

```diff
@@ -623,6 +623,45 @@ sub_library_info(struct xnvmec *XNVME_UNUSED(cli))
 	return err;
 }
 
+static int
+sub_dsm(struct xnvmec *cli)
+{
+	// We can only define one range in CLI
+	uint32_t nr = 1;
+	struct xnvme_spec_dsm_range dsm_range;
+
+	struct xnvme_dev *dev = cli->args.dev;
+	struct xnvme_cmd_ctx ctx = xnvme_cmd_ctx_from_dev(dev);
+	uint32_t nsid = cli->args.nsid;
+	uint64_t slba = cli->args.slba;
+	uint32_t nlb = cli->args.nlb;
+	bool ad = cli->args.ad;
+	bool idw = cli->args.idw;
+	bool idr = cli->args.idr;
+	int err;
+
+	xnvmec_pinf("xNVMe DSM");
+	if (!cli->given[XNVMEC_OPT_NSID]) {
+		nsid = xnvme_dev_get_nsid(cli->args.dev);
+	}
+
+	dsm_range = (struct xnvme_spec_dsm_range){
+		.cattr = 0,
+		.nlb = nlb,
+		.slba = slba,
+	};
+
+	err = xnvme_nvm_dsm(&ctx, nsid, nr, ad, idw, idr, &dsm_range);
+	if (err || xnvme_cmd_ctx_cpl_status(&ctx)) {
+		xnvmec_perr("xnvme_adm_dsm()", err);
```

Comment:
Seems like a typo here? Should ``xnvme_adm_dsm`` --> ``xnvme_nvm_dsm``?

Reply:
Fixed!